splitPoundsPennies.js

```diff
@@ -0,0 +1,22 @@
+const log4js = require('log4js');
+const logger = log4js.getLogger('logs\\debug.log');
+
+function splitPoundsPennies(allTransactions) {
+    for (let i=0;i<allTransactions.length;i++) {
+        let stringAmount=allTransactions[i].Amount.toString();
+        let arrayAmount=stringAmount.split('.');
+        allTransactions[i].amountPounds=arrayAmount[0];
+        allTransactions[i].amountPennies=arrayAmount[1];
+        if (allTransactions[i].amountPennies===undefined) {
+            allTransactions[i].amountPennies='0';

```

Comment:
Elsewhere `amountPennies` is a number; shouldn't this be `=0`?

Whether it should or not, this can be written neatly in one line as:

    transaction.amountPennies = arrayAmount[1] || 0;

(The righthand side being read as "arrayAmount one, or zero")

This works because when encountering the `||` operator, JS will go from left to right until it's able to work out the value of the expression as a whole; if `arrayAmount[1]` is truthy (exists and isn't zero) then it will stop there and use that because it knows the expression as a whole is truthy. If `arrayAmount[1]` is falsy (e.g. undefined or zero) then it will keep going, see zero and return that (since it's out of options).